base/test/testSignalGeneratorSrc_tests.cpp

```diff
@@ -2,76 +2,88 @@
     source->addOutputPin(metadata);
     auto sink = boost::shared_ptr<ExternalSinkModule>(new ExternalSinkModule());
     source->setNext(sink);
-    source->init();
-    sink->init();
+    BOOST_TEST(source->init());
+    BOOST_TEST(sink->init());
     source->step();
-    auto frame = sink->pop();
-    auto frameMetadata = frame.begin()->second->getMetadata();
-    BOOST_ASSERT(frameMetadata->getFrameType() == FrameMetadata::RAW_IMAGE_PLANAR);
-}
-
-BOOST_AUTO_TEST_CASE(statsink)
-{
-    auto source = boost::shared_ptr<TestSignalGenerator>(new TestSignalGenerator(TestSignalGeneratorProps()));
-    auto metadata = framemetadata_sp(new RawImagePlanarMetadata(640, 360, ImageMetadata::ImageType::YUV420, size_t(0), CV_8U));
-    source->addOutputPin(metadata);
-    auto sink = boost::shared_ptr<Module>(new StatSink());
-    source->setNext(sink);
-    boost::shared_ptr<PipeLine> p;
-    p = boost::shared_ptr<PipeLine>(new PipeLine("test"));
-    p->appendModule(source);
-    if (!p->init())
-    {
-        throw AIPException(AIP_FATAL, "Engine Pipeline init failed. Check IPEngine Logs for more details.");
-    }
-    p->run_all_threaded();
-    boost::this_thread::sleep_for(boost::chrono::seconds(2));
-    p->stop();
-    p->term();
-    p->wait_for_all();
-    p.reset();
+    auto frames = sink->try_pop();
+    BOOST_TEST(frames.size() == 1);
+    auto outputFrame = frames.cbegin()->second;
+    BOOST_TEST(outputFrame->getMetadata()->getFrameType() == FrameMetadata::RAW_IMAGE_PLANAR);
+    Test_Utils::saveOrCompare("./data/testOutput/TestSample.raw", const_cast<const uint8_t *>(static_cast<uint8_t *>(outputFrame->data())), outputFrame->size(), 0);
 }
 
-BOOST_AUTO_TEST_CASE(fileWriterSink)
+BOOST_AUTO_TEST_CASE(PropsChange)
 {
-    std::vector<std::string> Files = {"./data/testsample1.raw"};
-    Test_Utils::FileCleaner f(Files);
     auto source = boost::shared_ptr<TestSignalGenerator>(new TestSignalGenerator(TestSignalGeneratorProps(640, 360)));
     auto metadata = framemetadata_sp(new RawImagePlanarMetadata(640, 360, ImageMetadata::ImageType::YUV420, size_t(0), CV_8U));
     source->addOutputPin(metadata);
-
-    auto sink = boost::shared_ptr<FileWriterModule>(new FileWriterModule(FileWriterModuleProps("./data/testsample1.raw")));
+    auto sink = boost::shared_ptr<SinkModule>(new SinkModule(SinkModuleProps()));
     source->setNext(sink);
-
-    boost::shared_ptr<PipeLine> p;
-    p = boost::shared_ptr<PipeLine>(new PipeLine("test"));
-    p->appendModule(source);
-    if (!p->init())
-    {
-        throw AIPException(AIP_FATAL, "Engine Pipeline init failed. Check IPEngine Logs for more details.");
-    }
-    p->run_all_threaded();
-    boost::this_thread::sleep_for(boost::chrono::seconds(1));
-    p->stop();
-    p->term();
-    p->wait_for_all();
-    p.reset();
+    source->init();
+    sink->init();
+    source->step();
+    auto sinkQue = sink->getQue();
+    auto framedata = sinkQue->pop();
```

Comment:
You should call it frames not framedata. Its a frame_container (with potentially multiple frames). 
Rewrite this without using auto please.